pennylane/labs/resource_estimation/resource_config.py

```diff
@@ -0,0 +1,209 @@
+    ResourceRX,
+    ResourceRY,
+    ResourceRZ,
+)
+from .resource_operator import ResourceOperator
+from .templates import (
+    ResourceAliasSampling,
+    ResourceMPSPrep,
+    ResourcePrepTHC,
+    ResourceQROMStatePreparation,
+    ResourceQubitizeTHC,
+    ResourceQubitUnitary,
+    ResourceSelectPauliRot,
+    ResourceSelectTHC,
+)
+
+
+class DecompositionType(StrEnum):
+    """Specifies the type of decomposition to override."""
+
+    ADJOINT = "adj"
+    CONTROLLED = "ctrl"
+    POW = "pow"
+
+
+class ResourceConfig:
+    """A container to track the configuration for errors, precisions, and custom decompositions for the
+    resource estimation pipeline.
+    """
+
+    def __init__(self) -> None:
+        self.resource_op_precisions = {
+            ResourceRX: {"precision": 1e-9},
+            ResourceRY: {"precision": 1e-9},
+            ResourceRZ: {"precision": 1e-9},
+            ResourceCRX: {"precision": 1e-9},
+            ResourceCRY: {"precision": 1e-9},
+            ResourceCRZ: {"precision": 1e-9},
+            ResourceSelectPauliRot: {"precision": 1e-9},
+            ResourceQubitUnitary: {"precision": 1e-9},
+            ResourceQROMStatePreparation: {"precision": 1e-9},
+            ResourceMPSPrep: {"precision": 1e-9},
+            ResourceAliasSampling: {"precision": 1e-9},
+            ResourceQubitizeTHC: {"rotation_precision": 15, "coeff_precision": 15},
+            ResourceSelectTHC: {"rotation_precision": 15, "coeff_precision": 15},
+            ResourcePrepTHC: {"rotation_precision": 15, "coeff_precision": 15},
+        }
+        self._custom_decomps = {}
+        self._adj_custom_decomps = {}
+        self._ctrl_custom_decomps = {}
+        self._pow_custom_decomps = {}
+
+    def __str__(self):
+        decomps = [op.__name__ for op in self._custom_decomps]
+        adj_decomps = [f"Adjoint({op.__name__})" for op in self._adj_custom_decomps]
+        ctrl_decomps = [f"Controlled({op.__name__})" for op in self._ctrl_custom_decomps]
+        pow_decomps = [f"Pow({op.__name__})" for op in self._pow_custom_decomps]
+
+        all_op_strings = decomps + adj_decomps + ctrl_decomps + pow_decomps
+        op_names = ", ".join(all_op_strings)
+
+        dict_items_str = ",\n".join(
+            f"    {key.__name__}: {value!r}" for key, value in self.errors_and_precisions.items()
+        )
+
+        formatted_dict = f"{{\n{dict_items_str}\n}}"
+
+        return (
+            f"ResourceConfig(\n"
+            f"  precisions = {formatted_dict},\n"
+            f"  custom decomps = [{op_names}]\n)"
+        )
+
+    def __repr__(self) -> str:
+        return f"ResourceConfig(precisions = {self.errors_and_precisions}, custom_decomps = {self._custom_decomps}, adj_custom_decomps = {self._adj_custom_decomps}, ctrl_custom_decomps = {self._ctrl_custom_decomps}, pow_custom_decomps = {self._pow_custom_decomps})"
+
+    def set_single_qubit_rotation_error(self, error: float) -> None:
```

Comment:
🤷🏼 ,
```suggestion
    def set_single_qubit_rotation_precision(self, precision: float) -> None:
```
Bit more consistent.